python/tank/util/session.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Session management for Toolkit.
+"""
+
+import os
+
+from tank_vendor.shotgun_api3 import Shotgun
+from tank_vendor.shotgun_api3 import AuthenticationFault
+from ConfigParser import SafeConfigParser
+from .shotgun import get_associated_sg_config_data
+
+# Configure logging
```

Comment:
I added it to the Hot Topics.

From: Manne Öhrström <notifications@github.com<mailto:notifications@github.com>>
Reply-To: shotgunsoftware/tk-core <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, February 17, 2015 at 9:29 AM
To: shotgunsoftware/tk-core <tk-core@noreply.github.com<mailto:tk-core@noreply.github.com>>
Cc: JF Boismenu <jean.francois.boismenu@autodesk.com<mailto:jean.francois.boismenu@autodesk.com>>
Subject: Re: [tk-core] For #28030, [security] User/password sign in on tank command with session caching (#129)


In python/tank/util/session.py<https://github.com/shotgunsoftware/tk-core/pull/129#discussion_r24818505>:

> +# By accessing, using, copying or modifying this work you indicate your
> +# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
> +# not expressly granted therein are reserved by Shotgun Software Inc.
> +
> +"""
> +Session management for Toolkit.
> +"""
> +
> +import os
> +
> +from tank_vendor.shotgun_api3 import Shotgun
> +from tank_vendor.shotgun_api3 import AuthenticationFault
> +from ConfigParser import SafeConfigParser
> +from .shotgun import get_associated_sg_config_data
> +
> +# Configure logging


let's discuss how to best accommodate this logging!

—
Reply to this email directly or view it on GitHub<https://github.com/shotgunsoftware/tk-core/pull/129/files#r24818505>.